common/sc_popen.cpp

```diff
@@ -156,18 +156,49 @@ static SC_Lock processlist_mutex;
 #    define THREAD_LOCK() processlist_mutex.lock()
 #    define THREAD_UNLOCK() processlist_mutex.unlock()
 
+std::string quoteWindowsArg(const std::string& arg) {
+    if (arg.empty())
+        return "\"\"";
+
+    bool needQuotes = arg.find_first_of(" \t\"") != std::string::npos;
+    if (!needQuotes)
+        return arg;
+
+    std::string result = "\"";
```

Comment:
I think you have just reinvented  `std::quoted` :) https://en.cppreference.com/w/cpp/io/manip/quoted

Simple example: https://godbolt.org/z/W9G8xehEh

Reply:
Thanks! This is exactly the kind of feedback I was hoping for. I'll try it and run against the tests I added here.

BTW the basic assumption in tests is that echo on Windows outputs the argument exactly as it was passed to the command, including escape chars. Is this what a Windows app expects, when it's supposed handle an argument with spaces or quotes?